core/arch/arm/include/mm/core_memprot.h

```diff
@@ -106,4 +106,18 @@ void *phys_to_virt_io(paddr_t pa);
  */
 paddr_t virt_to_phys(void *va);
 
+/*
+ * Return runtime usable address, irrespective of whether
+ * the MMU is enabled or not.
+ */
+vaddr_t core_mmu_get_va(paddr_t pa, enum teecore_memtypes type);
+
+/*
+ * Return runtime usable io address, irrespective of whether
+ * the MMU is enabled or not.
+ * @base has to be in a memory area of type MEM_AREA_IO_SEC or
+ * MEM_AREA_IO_NSEC. Returns NULL on failure.
```

Comment:
`base` should be `pa`.

Suggestion for the description, inspired from from [`phys_to_virt_io()`](https://github.com/OP-TEE/optee_os/blob/master/core/arch/arm/include/mm/core_memprot.h#L97-L99))
```Patch
- * @base has to be in a memory area of type MEM_AREA_IO_SEC or
- * MEM_AREA_IO_NSEC. Returns NULL on failure.
+ * Translate physical address to virtual address trying MEM_AREA_IO_SEC
+ * first then MEM_AREA_IO_NSEC if not found.
+ * Returns NULL on failure or a valid virtual address on success.


```